tools/blue_deployment/helpers.js

```diff
@@ -0,0 +1,136 @@
+const colors = require('colors/safe');
+
+/**
+ * Return a promise that resolves once a transaction has been mined.
+ *
+ * @param {object} web3 The web3 instance to use.
+ * @param {string} txHash The transaction hash of the transaction to wait for.
+ * @param {number} [interval=500] The interval of how often we want to check
+ *                 the transaction status.
+ */
+const waitForTransactionReceiptMined = (web3, txHash, interval = 500) => {
+    const transactionReceiptAsync = (resolve, reject) => {
+        web3.eth.getTransactionReceipt(txHash, (error, receipt) => {
+            if (error) {
+                reject(error);
+            } else if (receipt == null) {
+                setTimeout(
+                    () => transactionReceiptAsync(resolve, reject),
+                    interval,
+                );
+            } else {
+                resolve(receipt);
+            }
+        });
+    };
+
+    if (Array.isArray(txHash)) {
+        return Promise.all(txHash.map(
+            oneTxHash => waitForTransactionReceiptMined(web3, oneTxHash, interval),
+        ));
+    }
+    if (typeof txHash === 'string') {
+        return new Promise(transactionReceiptAsync);
+    }
+    throw new Error('Invalid Type: ' + txHash);
+};
+
+/**
+ * A Signer that assumes that the accounts used to sign the messages are
+ * permanently unlocked in the web3 instance it points to.
+ */
+class UnlockedWeb3Signer {
+    /**
+     * @param {object} web3 The web3 instance to use.
+     */
+    constructor(web3) {
+        this.web3 = web3;
+
+        this.signTransaction = this.signTransaction.bind(this);
+    }
+
+    /**
+     * @param {object} transactionObject The transaction object to sign.
+     *                 See {@link https://web3js.readthedocs.io/en/1.0/web3-eth.html#sendtransaction}
+     */
+    signTransaction(transactionObject) {
+        return this.web3.eth.signTransaction(transactionObject);
+    }
+}
+
+/**
+ * Deploy the provided deployment objects, as provided by
+ * {@link ContractRegistry#toLiveTransactionObjects}, in sequence.
+ *
+ * @param {object} signer The signer used to sign the transactions.
+ *                 See {@link UnlockedWeb3Signer}.
+ * @param {object} sendWeb3 The web3 instance used to send the deployment transactions.
```

Comment:
Why `sendWeb3` instead of just `web3`? Is there another web3?

Reply:
The `signer` was originally a `signWeb3`. But yes, should be renamed now.